server/app/Icepeak/Main.hs

```diff
@@ -52,20 +53,20 @@ main = do
   config <- execParser (configInfo env)
 
   -- start logging as early as possible
-  logger <- Logger.newLogger (fromIntegral $ configQueueCapacity config)
+  logger <- Logger.newLogger config
   loggerThread <- Async.async $ Logger.processLogRecords logger
+  SentryLogging.runWithCrashLogger "Control loop error" (Logger.loggerSentryService logger) $ do
+    -- setup metrics if enabled
+    icepeakMetrics <- forM (configMetricsEndpoint config) $ const $ do
+      void $ Prometheus.register Prometheus.Metric.GHC.ghcMetrics
+      Metrics.createAndRegisterIcepeakMetrics
 
-  -- setup metrics if enabled
-  icepeakMetrics <- forM (configMetricsEndpoint config) $ const $ do
-    void $ Prometheus.register Prometheus.Metric.GHC.ghcMetrics
-    Metrics.createAndRegisterIcepeakMetrics
+    eitherCore <- Core.newCore config logger icepeakMetrics
+    either (postLog logger LogInfo . Text.pack) runCore eitherCore
 
-  eitherCore <- Core.newCore config logger icepeakMetrics
-  either putStrLn runCore eitherCore
-
-  -- only stop logging when everything else has stopped
-  Logger.postStop logger
-  Async.wait loggerThread
+    -- only stop logging when everything else has stopped
+    Logger.postStop logger
+    Async.wait loggerThread
```

Comment:
Given my remark above, stopping the logger should then be outside of the scope of the handler above, so that we don't end up in a situation where that unexpectedly fails and then tries to post something to the logger that has already been stopped.